core/baseline-core-v1.0-psd01.md

```diff
@@ -2792,105 +2792,156 @@ As has been done throughout this document, there are BPI layer-specific security
+
 ## 7.3. BPI Data Orchestration
 
 To accommodate a high-volume, low latency environment with many data changes, BPI Data Orchestration has the following requirements:
 
 #### **[R289]**	
 Data Orchestration utilized in a BPI MUST NOT be a single point of failure.
 
+[[R289]](#r289) Testability: Data Orchestration solutions without a single point of failure are distributed. An example of such an implementation with a complete test suite is [Apache Kafka](https://github.com/apache/kafka). Therefore, the requirement is testable.
+
 #### **[R290]**	
 Data Orchestration utilized in a BPI MUST preserve source consistency. 
 
+[[R290]](#r290) Testability: Data Orchestration solutions preserving source consistency typically utilize a queued, redundant publish/subscribe messaging model. An example of such an implementation with a complete test suite is [Apache Kafka](https://github.com/apache/kafka). Therefore, the requirement is testable.
+
 *To avoid subscribers seeing partial and/or inconsistent data, BPI Data Orchestration has the following requirements:*
 
 #### **[R291]**	
 Data Orchestration utilized in a BPI MUST implement transaction boundaries.
 
 *This means that a single user's action can trigger atomic updates.*
 
+[[R291]](#r291) Testability: Data Orchestration solutions with transaction boundaries typically utilize topic-driven publish/subscribe messaging models. An example of such an implementation with a complete test suite is [Apache Kafka](https://github.com/apache/kafka). Therefore, the requirement is testable.
+
 #### **[R292]**	
-Data Orchestration utilized in a BPI MUST Commit the exact order in which operations happened on the primary database.
+Data Orchestration utilized in a BPI MUST commit the exact order in which operations happened on the primary database.
+
+[[R292]](#r292) Testability: Data Orchestration solutions which commit to and preserve operations order utilize a publish/subscribe messaging model where each message is uniquely associated to a topic and where the message identifier is deterministically created. An example of such an implementation with a complete test suite is [Apache Kafka](https://github.com/apache/kafka). Therefore, the requirement is testable.
 
 #### **[R293]**	
 Data Orchestration utilized in a BPI MUST support a consistent state.
 
+[[R293]](#r293) Testability: Data Orchestration solutions that support a consistent state typically utilize a queued publish/subscribe messaging model where each message is uniquely associated to a topic and where the message identifier is deterministically created to preserve state by maintaining a history of state changes. An example of such an implementation with a complete test suite is [Apache Kafka](https://github.com/apache/kafka). Therefore, the requirement is testable.
```

Comment:
It isn't clear how maintaining a history of state changes supports consistent state in a distributed system - I assume the key to that is the reconciliation process, which isn't mentioned in the text as written.

Reply:
@chaals no, the key is the properly ordered state changes which can be done because you have a deterministic message identifier ordering the state changes across topics which group state. It says consistent state, that does not have to be correct. Just that changes are properly ordered.

Reply:
There is one half-sentence missing, "..., and the message is removed from the data orchestration solution once consumed from a given topic, ..." that should clarify it. @chaals